north/src/runtime/console.rs

```diff
@@ -234,4 +278,42 @@ async fn connection(stream: TcpStream, mut tx_main: EventTx) {
             break;
         }
     }
+
+    async fn create_installation_event<R: Unpin + Read>(
+        mut reader: R,
+        tmp_installation_dir: &std::path::Path,
+        msg_id: MessageId,
+        size: usize,
+        install_file_name: String,
+        tx: sync::Sender<api::Message>,
+    ) -> io::Result<Event> {
+        const BUF_SIZE: usize = 4096;
+        let mut buffer = [0u8; BUF_SIZE];
+        // now we will receive the file with [size] bytes
+        let tmp_installation_file_path = tmp_installation_dir.join(&install_file_name);
+        let mut tmpfile = OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(&tmp_installation_file_path)
+            .await?;
+        let mut received_bytes = 0usize;
+        while received_bytes < size {
+            let received = reader.read(&mut buffer).await?;
+            if received > 0 {
+                tmpfile.write_all(&buffer[..received]).await?;
+                received_bytes += received;
+            }
+        }
```

Comment:
No need to mess with a buffer as in the old times. This here is:
```rust
io::copy(reader.take(size), tmp file).await?
```

Reply:
thanks...much nicer